Source/dom/style/Style.js

```diff
@@ -181,3 +199,56 @@ Style.define('innerShadows', {
     this._object.setInnerShadows(objects)
   },
 })
+
+Style.define('sharedStyleId', {
+  get() {
+    if (this._object.sharedObjectID()) {
+      return String(this._object.sharedObjectID())
+    }
+    return null
+  },
+  set(newId) {
+    const documentData = this._object.documentData()
+    const container = documentData.sharedObjectContainerOfType(
+      this._object.type()
+    )
+
+    if (!newId) {
+      const currentSharedId = this._object.sharedObjectID()
+      if (currentSharedId) {
+        const currentSharedObject = container.sharedStyleWithID(currentSharedId)
+        currentSharedObject.unregisterInstance(this._object)
+      }
+    } else {
+      const sharedStyle = container.sharedStyleWithID(newId)
+      if (!sharedStyle) {
+        throw new Error('Seems like this shared style does not exists')
+      }
+      container.registerInstance_withSharedStyle(this._object, sharedStyle)
+    }
+  },
+})
+
+Style.define('sharedStyle', {
+  exportable: false,
+  enumerable: false,
+  get() {
+    const { sharedStyleId } = this
+    if (!sharedStyleId) {
+      return null
+    }
+    const documentData = this._object.documentData()
+    const container = documentData.sharedObjectContainerOfType(
+      this._object.type()
+    )
+    return wrapObject(container.sharedStyleWithID(sharedStyleId))
```

Comment:
I'm not entirely sure I'm following what kind of object this code is wrapping, but as above we want to avoid accessing the `sharedObjectContainers`. But I suspect you could again use `MSLayer+SharedStyles.h` and just return  `_this.sharedStyle`

Reply:
hum I don't have access to the Layer here. And worse, the style might not be inserted in a Layer but might already reference a shared style.